File: apps/sdkwork-rtc-h5/src/bootstrap/environment.ts
export interface RtcEnvironment {
  apiBaseUrl: string;
  appbaseAppApiBaseUrl: string;
  backendApiBaseUrl: string;
  appbaseLoginUrl: string;
  defaultMediaMode: "audio" | "video" | "live";
  providerSelection: string;
  mobile: {
    maxParticipants: number;
    audioOnlyFallback: boolean;
  };
}

function normalizeBaseUrl(value: string | undefined, fallback: string): string {
  const normalized = String(value ?? "").trim();
  return normalized || fallback;
}

function deriveAppApiBaseUrl(applicationPublicHttpUrl: string): string {
  return `${applicationPublicHttpUrl.replace(/\/+$/u, "")}/app/v3/api`;
}

function deriveBackendApiBaseUrl(applicationPublicHttpUrl: string): string {
  return `${applicationPublicHttpUrl.replace(/\/+$/u, "")}/backend/v3/api`;
}

export function resolveEnvironment(): RtcEnvironment {
  const applicationPublicHttpUrl = normalizeBaseUrl(
    import.meta.env.VITE_SDKWORK_RTC_H5_APPLICATION_PUBLIC_HTTP_URL,
    "http://127.0.0.1:18088",
  );

  return {
    apiBaseUrl: normalizeBaseUrl(
      import.meta.env.VITE_SDKWORK_RTC_H5_APP_API_BASE_URL,
      deriveAppApiBaseUrl(applicationPublicHttpUrl),
    ),
    backendApiBaseUrl: normalizeBaseUrl(
      import.meta.env.VITE_SDKWORK_RTC_H5_BACKEND_API_BASE_URL,
      deriveBackendApiBaseUrl(applicationPublicHttpUrl),
    ),
    appbaseLoginUrl: normalizeBaseUrl(
      import.meta.env.VITE_SDKWORK_RTC_H5_APPBASE_LOGIN_URL,
      "http://127.0.0.1:3900",
    ),
    defaultMediaMode: "video",
    providerSelection: "auto",
    mobile: {
      maxParticipants: 9,
      audioOnlyFallback: true,
    },
  };
}

File: apps/sdkwork-rtc-flutter-mobile/lib/bootstrap/environment.dart
class RtcEnvironment {
  final String apiBaseUrl;
  final String backendApiBaseUrl;
  final String platformApiGatewayHttpUrl;
  final String appbaseLoginUrl;
  final String defaultMediaMode;
  final String providerSelection;
  final int maxParticipants;
  final bool audioOnlyFallback;

  const RtcEnvironment({
    required this.apiBaseUrl,
    required this.backendApiBaseUrl,
    required this.platformApiGatewayHttpUrl,
    required this.appbaseLoginUrl,
    this.defaultMediaMode = 'video',
    this.providerSelection = 'auto',
    this.maxParticipants = 9,
    this.audioOnlyFallback = true,
  });
}

String _normalizeBaseUrl(String value, String fallback) {
  final normalized = value.trim();
  return normalized.isEmpty ? fallback : normalized;
}

String _deriveAppApiBaseUrl(String applicationPublicHttpUrl) {
  final normalized = applicationPublicHttpUrl.replaceAll(RegExp(r'/+$'), '');
  return '$normalized/app/v3/api';
}

String _deriveBackendApiBaseUrl(String applicationPublicHttpUrl) {
  final normalized = applicationPublicHttpUrl.replaceAll(RegExp(r'/+$'), '');
  return '$normalized/backend/v3/api';
}

RtcEnvironment resolveEnvironment() {
  const applicationPublicHttpUrl = String.fromEnvironment(
    'SDKWORK_RTC_APPLICATION_PUBLIC_HTTP_URL',
    defaultValue: 'http://127.0.0.1:18088',
  );

  return RtcEnvironment(
    apiBaseUrl: _normalizeBaseUrl(
      const String.fromEnvironment('SDKWORK_RTC_APP_API_BASE_URL'),
      _deriveAppApiBaseUrl(applicationPublicHttpUrl),
    ),
    backendApiBaseUrl: _normalizeBaseUrl(
      const String.fromEnvironment('SDKWORK_RTC_BACKEND_API_BASE_URL'),
      _deriveBackendApiBaseUrl(applicationPublicHttpUrl),
    ),
    platformApiGatewayHttpUrl: _normalizeBaseUrl(
      const String.fromEnvironment('SDKWORK_RTC_PLATFORM_API_GATEWAY_HTTP_URL'),
      'http://127.0.0.1:3900',
    ),
    appbaseLoginUrl: _normalizeBaseUrl(
      const String.fromEnvironment('SDKWORK_RTC_APPBASE_LOGIN_URL'),
      _normalizeBaseUrl(
        const String.fromEnvironment('SDKWORK_RTC_PLATFORM_API_GATEWAY_HTTP_URL'),
        'http://127.0.0.1:3900',
      ),
    ),
  );
}
